src/BMPEncoder.ts

```diff
@@ -4,74 +4,99 @@ import { BITMAPV5HEADER } from './constants';
 
 export interface ImageCodec {
   /**
-   * Image bit depth.
-   */
-  bitDepth: number;
-  /**
-   * Image height.
+   * Image height. Data offset: 18
    */
   height: number;
   /**
-   * Image width.
+   * Image width. Data offset: 22
    */
   width: number;
   /**
-   * Image data.
+   * Image number of channels. Calculated with decoded data.
    */
-  data: IOBuffer | ArrayBufferLike | ArrayBufferView | Buffer;
+  channels: number;
   /**
-   * Image number of channels.
+   * Image number of channels excluding alpha. Calculated with decoded data.
    */
-  channels: number;
+  components?: number;
   /**
-   * Image number of channels excluding alpha.
+   * Image bit depth. Data offset: 30
    */
-  components: number;
+  bitsPerPixel: number;
   /**
-   * Horizontal number of pixels per meter.
+   * Image compression type. Data offset: 34
+   */
+  compression?: number;
+  /**
+   * Defines which bits represent which color. Data offset: 58 to 66
+   */
+  colorMasks?: number[];
+  /**
+   * Horizontal number of pixels per meter. Data offset: 38
    */
   xPixelsPerMeter?: number;
   /**
-   * Vertical number of pixels per meter.
+   * Vertical number of pixels per meter. Data offset: 42
    */
   yPixelsPerMeter?: number;
+  /**
+   * Image data. Data offset: 146.
```

Comment:
As a user of the library, how am I supposed to interpret the offset values / do I care? I think this information should be internal an not exposed in the documentation.